pramen-py/src/pramen_py/metastore/reader.py

```diff
@@ -49,16 +50,87 @@ class MetastoreReader(MetastoreReaderBase):
 
     HOCON_METASTORE_TABLES_VARIABLE = "pramen.metastore.tables"
 
+    @staticmethod
     def _apply_uppercase_to_columns_names(
-        self, df: DataFrame, uppercase_columns: bool
+        df: DataFrame, uppercase_columns: bool
     ) -> DataFrame:
         if uppercase_columns:
             return df.select([F.col(c).alias(c.upper()) for c in df.columns])
         else:
             return df
 
+    @staticmethod
+    def _extract_date_from_path(
+        path: str, target_partition_name: str, date_format: str
+    ) -> Optional[datetime.date]:
+        def is_file_hidden(column_name_: str, date_: str) -> bool:
+            if column_name_.startswith(("_", ".")):
+                return True
+            if date_.startswith(("_", ".")):
+                return True
+            return False
+
+        file_name = path.replace("\\", "/").rsplit("/", 1)[-1]
+        if not file_name:
+            return None
+
+        name_parts = file_name.split("=")
+        if not len(name_parts) == 2:
+            return None
+
+        partition_name, date = name_parts
+        if is_file_hidden(partition_name, date):
+            return None
+
+        if target_partition_name != partition_name:
+            raise ValueError(
+                f"Partition name mismatch for path: {path}\n"
+                f"Expected '{target_partition_name}' instead '{partition_name}'"
+            )
+
+        try:
+            return convert_str_to_date(date, fmt=date_format)
+        except (KeyError, ValueError):
+            raise ValueError(
+                f"Date format mismatch for path: {path}"
+                f"Expected '{date_format}' instead '{date}'"
+            )
+
+    def validate_partitions_paths(
+        self,
+        files_paths: List[str],
+        table: MetastoreTable,
+    ) -> None:
+        extracted_dates = []
+        for path in files_paths:
+            date = self._extract_date_from_path(
+                path,
+                table.info_date_settings.column,
+                table.info_date_settings.format,
+            )
+            if date:
+                extracted_dates.append(date)
+        if not extracted_dates:
+            raise ValueError(
+                f"The directory does not contain partitions by "
+                f"'{table.info_date_settings.column}': {table.path}"
+            )
+
+    @staticmethod
+    def _filter_info_dates_by_until(
+        dates: List[datetime.date],
+        until: datetime.date,
+    ) -> List[datetime.date]:
+        def before_until(date: datetime.date) -> bool:
+            return date <= until
+
+        return list(filter(before_until, dates))
+
     def _read_table(self, table_format: TableFormat, path: str) -> DataFrame:
-        return self.spark.read.format(table_format.value).load(path)
+        try:
+            return self.spark.read.format(table_format.value).load(path)
+        except AnalysisException:
+            raise Exception(f"Unable to access directory: {path}")
```

Comment:
You mentioned you wanted to improve the error message by adding the cause of the error (from the original analysts exception)?

Reply:
Original exception is outing by this way